src/main/java/com/google/api/codegen/csharp/CSharpSnippetSetRunner.java

```diff
@@ -48,23 +49,20 @@ public GeneratedResult generate(
             snippetFileName,
             ImmutableMap.<String, Object>of("context", context));
 
-    // TODO don't depend on a cast here
-    CSharpGapicContext csharpContext = (CSharpGapicContext) context;
-
     String outputFilename = snippets.generateFilename(element).prettyPrint();
+    CSharpContextCommon csharpContextCommon = new CSharpContextCommon();
```

Comment:
I confess I have very little idea what these different contexts are about at this stage.